python/tchannel/tornado/connection.py

```diff
@@ -527,21 +528,29 @@ def _stream(self, context, message_factory):
         :param context: Request or Response object
         """
         args = []
-        for i, argstream in enumerate(context.argstreams):
-            chunk = yield argstream.read()
-            args.append(chunk)
-            chunk = yield argstream.read()
-            while chunk:
-                message = (message_factory.
-                           build_raw_message(context, args))
-                yield self.write(message, context.id)
-                args = [chunk]
+        try:
+            for argstream in context.argstreams:
                 chunk = yield argstream.read()
-
-        # last piece of request/response.
-        message = (message_factory.
-                   build_raw_message(context, args, is_completed=True))
-        yield self.write(message, context.id)
+                args.append(chunk)
+                chunk = yield argstream.read()
+                while chunk:
+                    message = (message_factory.
+                               build_raw_message(context, args))
+                    yield self.write(message, context.id)
+                    args = [chunk]
+                    chunk = yield argstream.read()
+
+            # last piece of request/response.
+            message = (message_factory.
+                       build_raw_message(context, args, is_completed=True))
+            yield self.write(message, context.id)
+            context.state = StreamState.completed
+        # Stop streamming immediately if exception occurs on the handler side
+        except TChannelException:
+            # raise by tchannel intentionally
+            pass
```

Comment:
What do you mean? Isn't this the exception we use for all custom exceptions defined in this project?

Also, regardless, usually you don't want to silently blackhole exceptions.